docs/root/intro/arch_overview/upstream/upstream_filters.rst

```diff
@@ -4,8 +4,10 @@ Upstream network filters
 ========================
 
 Upstream clusters provide an ability to inject network level (L3/L4)
-:ref:`filters <arch_overview_network_filters>`. The filters apply to the
-connection to the upstream hosts, using the same API presented by listeners for
+filters. It should be noted that network filter needs to
```

Comment:
```suggestion
filters. It should be noted that a network filter needs to
```

Reply:
Fixed. Ran this text through a grammar checker, just to make sure that I didn't miss anything else.